pandas/io/pytables.py

```diff
@@ -4875,8 +4876,16 @@ def _unconvert_index(
 
 
 def _maybe_convert_for_string_atom(
-    name: str, block, existing_col, min_itemsize, nan_rep, encoding, errors
+    name: str,
+    block,
+    existing_col,
+    min_itemsize,
+    nan_rep,
+    encoding,
+    errors,
+    block_columns: List[str] = [],
```

Comment:
remove the default arg, the annotation is enough

Reply:
the default `[]` is used to keep the interface consistency.
but since the function is only called once I think it's okay to remove, maybe
will remove it.

Reply:
yes and *never* should you use mutable arguments